### DISCUSSION TITLE
[Bug]: Could not delete item

### DISCUSSION THREAD

**Discussion by Dau-Duc-Hai** [pimcore/pimcore#11115] [Q&A / Support] [answered]
### Expected behavior

Cannot delete assets. Find out that, thumbnails (in var/tmp/ folder) are not apache (www-data) owner

### Actual behavior

System alert cannot delete then delete folder instead

### Steps to reproduce

Try to delete assets

**Comment by brusch** [comment]
Fix your permissions ...

**Comment by Dau-Duc-Hai** [comment]
Can you show me? the website / apache run under www-data and I thinks www-data should the account to generate thumbnails? Please help

**Comment by maralkumbe** [comment]
Hi, I think I have the same problem here (pimcore version 10.2.7 inside docker environment https://github.com/pimcore/skeleton/ ):

{"success":false,"message":"Unable to delete file located at: \/var\/www\/html\/public\/var\/tmp\/thumbnails\/TERRITORIO\/test\/image-thumb__332__-low-quality-preview.svg. unlink(\/var\/www\/html\/public\/var\/tmp\/thumbnails\/TERRITORIO\/test\/image-thumb__332__-low-quality-preview.svg): Permission denied"}

If I change permissions on the folder with 
chmod -R 777 /var/www/html/public/var/tmp
or 
chown -R www-data:www-data 
then it works once, but at the next save the new files get saved again as root:root 755 and the problem reappears.

I can't find solutions in the legacy forum, this forum and this page:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Permissions.html

The problem I see is files get saved on the file system as root:root (but deleted as www-data I suppose as it only works if www-data is able to delete).

```
root@9fe61305ed1a:/var/www/html/public/var/tmp/thumbnails# ls -la
total 48
drwxrwxrwx.   6 www-data www-data    82 Jan 29 13:36 .
drwxrwxrwx.   5 www-data www-data    56 Jan  4 13:39 ..
drwxr-xr-x.   4 root     root       142 Feb  1 00:36 TERRITORIO
...
```
Is it normal files get saved as root in this folder?

**Comment by maralkumbe** [comment, marked as answer]
Solved!

I forgot to set the userid:groupid in the docker-compose.yml file (as suggested as comment), on my server 33:33, in docker-compose.yml:

```
    php-fpm:
        user: '33:33' # set to your uid:gid
...
    php-fpm-debug:
        user: '33:33' # set to your uid:gid
...
    supervisord:
        user: '33:33' # set to your uid:gid
...
```
So the 3 above containers on the host save files as www-data and the nginx container now can modify those files.

**Reply by maralkumbe** [reply]
Hi, I fixed my previous answer, no root inside the container..., now that the container starts with uid 33 if I run a shell inside the php-fpm container that comes FROM the pimcore/pimcore:PHP8.0-fpm image I see the www-data user is 33:
```
# docker exec -it myproject_php-fpm_1 /bin/bash
www-data@54705cbf1e50:~/html$ grep www /etc/passwd
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
```
So I suppose "user: '33:33'" should work indipendently from the docker host environment and I'm asking myself why it is not set as default in docker-compose.yml and the meaning of the comment "set to _your_ uid:gid".
Thank you.

**Reply by shades3002** [reply]
Adding the user to all three containers works for me
